app/src/main/java/com/sunasterisk/music_72/screen/fragment/tracks/TracksViewModel.kt

```diff
@@ -3,17 +3,61 @@ package com.sunasterisk.music_72.screen.fragment.tracks
 import androidx.lifecycle.LiveData
 import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
+import com.sunasterisk.music_72.data.anotation.GenreName
 import com.sunasterisk.music_72.data.model.Track
 import com.sunasterisk.music_72.data.source.repository.TrackRepositoryImplementor
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.schedulers.Schedulers
+import java.util.*
 
 class TracksViewModel(private val repository: TrackRepositoryImplementor) : ViewModel() {
     private val _tracks = MutableLiveData<List<Track>>()
     val tracks: LiveData<List<Track>>
         get() = _tracks
+    private val _error = MutableLiveData<Throwable>()
+    val error: LiveData<Throwable>
+        get() = _error
+    private val compositeDisposable = CompositeDisposable()
+
+    fun getTracks(genre: String) {
+        if (genre === GenreName.ALL_TRACK) getAllTrack()
+        else getTracksByGenre(genre)
+    }
+
+    private fun getTracksByGenre(genre: String) {
+        val disposable = repository.getTracksByGenre(genre.toLowerCase(Locale.getDefault()), LIMIT_ITEM)
+            .subscribeOn(Schedulers.io())
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribe({
+                _tracks.value = it
+            }, {
+                _error.value = it
+            })
+        compositeDisposable.add(disposable)
```

Comment:
https://dev.to/ryugoo/how-to-make-reusable-compositedisposable-and-kotlin-extension-1db7
Đọc bài trên nhé 
```suggestion
        repository.getTracksByGenre(genre.toLowerCase(Locale.getDefault()), LIMIT_ITEM)
            .subscribeOn(Schedulers.io())
            .observeOn(AndroidSchedulers.mainThread())
            .subscribe({
                _tracks.value = it
            }, {
                _error.value = it
            }).addTo(compositeDisposable)
```